plugins/pip/venvShellHook.sh

```diff
@@ -1,8 +1,92 @@
-#!/bin/sh
+set -eu
+STATE_FILE="$DEVBOX_PROJECT_ROOT/.devbox/venv_check_completed"
+echo $STATE_FILE
 
-if ! [ -d "$VENV_DIR" ]; then
-    echo "Creating new venv environment in path: '${VENV_DIR}'"
-    python3 -m venv "$VENV_DIR"
-    echo "You can activate the virtual environment by running '. \$VENV_DIR/bin/activate' (for fish shell, replace '.' with 'source')" >&2
+is_valid_venv() {
+    [ -f "$1/bin/activate" ] && [ -f "$1/bin/python" ]
+}
+
+# Function to check if Python is a symlink to a Devbox Python
+is_devbox_python() {
```

Comment:
There's also the `test -ef` flag to check if two files are the same. So you could do:

```sh
if [ "$1/bin/python" -ef "$DEVBOX_PACKAGES_DIR/bin/python" ]; then
```